src/spec/datatypes.rs

```diff
@@ -0,0 +1,388 @@
+
+    /// Elements can be either optional or required.
+    pub element_required: bool,
+
+    /// Datatype
+    pub element: Box<Type>,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
+#[serde(rename = "map", rename_all = "kebab-case", tag = "type")]
+/// A map is a collection of key-value pairs with a key type and a value type.
+/// Both the key field and value field each have an integer id that is unique in the table schema.
+/// Map keys are required and map values can be either optional or required.
+/// Both map keys and map values may be any type, including nested types.
+pub struct MapType {
+    /// Key Id that is unique in table schema
+    pub key_id: i32,
+    /// Datatype of key
+    pub key: Box<Type>,
+    /// Value Id that is unique in table schema
+    pub value_id: i32,
+    /// If value is optional or required
+    pub value_required: bool,
+    /// Datatype of value
+    pub value: Box<Type>,
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn decimal() {
+        let record = r#"
+        {
+            "type": "struct",
+            "fields": [
+                {
+                    "id": 1,
+                    "name": "id",
+                    "required": true,
+                    "type": "decimal(9,2)"
+                }
+            ]
+        }
+        "#;
+
+        let result: StructType = serde_json::from_str(record).unwrap();
+        assert_eq!(
+            Type::Primitive(PrimitiveType::Decimal {
+                precision: 9,
+                scale: 2
+            }),
+            result.fields[0].field_type
+        );
+        let result_two: StructType = serde_json::from_str(
+            &serde_json::to_string(&result).expect("Failed to serialize result"),
+        )
+        .expect("Failed to serialize json");
+        assert_eq!(result, result_two);
+    }
+
+    #[test]
+    fn fixed() {
+        let record = r#"
+        {
+            "type": "struct",
+            "fields": [
+                {
+                    "id": 1,
+                    "name": "id",
+                    "required": true,
+                    "type": "fixed[8]"
+                }
+            ]
+        }
+        "#;
+
+        let result: StructType = serde_json::from_str(record).unwrap();
+        assert_eq!(
+            Type::Primitive(PrimitiveType::Fixed(8)),
+            result.fields[0].field_type
+        );
+        let result_two: StructType = serde_json::from_str(
+            &serde_json::to_string(&result).expect("Failed to serialize result"),
+        )
+        .expect("Failed to serialize json");
+        assert_eq!(result, result_two);
+    }
+
+    #[test]
+    fn struct_type() {
+        let record = r#"
+        {
+            "type": "struct",
+            "fields": [ {
+            "id": 1,
+            "name": "id",
+            "required": true,
+            "type": "uuid",
+            "initial-default": "0db3e2a8-9d1d-42b9-aa7b-74ebe558dceb",
```

Comment:
This is interesting, but not wrong. The physical type of a `uuid` is a fixed in Iceberg (this differs from Avro, where it is a string).